### PR TITLE
Migrate to Android X 

### PR DESCRIPTION
**Pull request by scottyab**
Addresses #362

* Migrate to Android X. 
* Build and Target version to 28
* source and target compatibility to java 1.8
* Gradle plugin updated to 3.4.2

I also bumped the version to 4.0.0 given this will likely be a breaking change for some users of the library.

### CODE REVIEW

**Comment by scottyab** [comment]
Likely some Circle CI config needs updating to use compile SDK 28 and newer buildtools but wasn't 100% sure on this.

File: gradle.properties [hunk 1]
```diff
@@ -35,4 +28,6 @@ POM_LICENCE_NAME=Apache v2
 POM_LICENCE_URL=https://github.com/splitwise/TokenAutoComplete/blob/master/LICENSE
 POM_LICENCE_DIST=repo
 POM_DEVELOPER_ID=mgod
-POM_DEVELOPER_NAME=Marshall Weir
\ No newline at end of file
+POM_DEVELOPER_NAME=Marshall Weir
+android.useAndroidX=true
+android.enableJetifier=true
```
Comment (mgod) [inline gradle.properties]: I think these should be removable now?
```
android.useAndroidX=true
android.enableJetifier=true
```

Comment (scottyab) [reply gradle.properties]: AFAIK we need `useAndroidX=true` ? but good spot `android.enableJetifier` should be false

**Comment by cketti** [comment]
@mgod Is there anything that still needs to be done before this PR can be merged? I'm happy to help with whatever work is necessary.

**Comment by mgod** [comment]
Nope, sorry. Just intend to have this as part of a 4.0 release with a couple of other breaking changes, so planning got a little bigger than just this change.

**Comment by mgod** [comment]
I'll put together a `4.0-beta01` release for this so you can get it running until I have time to review other details

**Comment by mgod** [comment]
You can now grab `com.splitwise:tokenautocomplete:4.0.0-beta01@aar`. I'm also considering migrating to Kotlin for everything. Would this be a problem for you?

**Comment by cketti** [comment]
> I'm also considering migrating to Kotlin for everything. Would this be a problem for you?

I'm proposing to use emoji reactions for the answer to this question.

:+1: Not a problem for me. Kotlin all the things!
:-1: I'd prefer the library sticking to Java

Personally, I'd welcome a switch to Kotlin.

**Pull request merged**